src/main/resources/pipeline/azure/azure.yaml.mustache

```diff
@@ -10,79 +10,62 @@ variables:
-          inputs:
-            gradleWrapperFile: gradlew
-            tasks: 'clean build jacocoMergedReport'
-            publishJUnitResults: true
-            workingDirectory: .
-            testResultsFiles: '**/build/test-results/test/TEST-*.xml'
-            sonarQubeRunAnalysis: true
-            sonarQubeServiceEndpoint: SonarqubeGradle
-            sonarQubeProjectName: '$(Build.Repository.Name)'
-            sonarQubeProjectKey: '$(Build.Repository.Name)'
-            sonarQubeProjectVersion: '$(Build.BuildNumber)'
-            sonarQubeGradlePluginVersion: 2.7
-            sonarQubeFailWhenQualityGateFails: false
-
-        - task: PublishCodeCoverageResults@1
-          displayName: 'Publish code coverage'
-          inputs:
-            summaryFileLocation: '$(System.DefaultWorkingDirectory)/build/reports/jacocoMergedReport/jacocoMergedReport.xml'
+  - job: BuildJob
+    displayName: 'Build'
+    pool:
+      name: Build
+      demands:
+        - java
+    steps:
+      - task: SonarQubePrepare@4
+        displayName: 'Prepare analysis on SonarQube'
+        inputs:
+          SonarQube: SonarQube
+          scannerMode: Other
+          extraProperties: |
+            sonar.projectKey=$(Build.Repository.Name)
+            sonar.projectName=$(Build.Repository.Name)
+            sonar.project.version=$(Build.BuildNumber)
+            sonar.branch.name=$(Build.SourceBranchName)
 
-        - task: SonarSource.sonarqube.6D01813A-9589-4B15-8491-8164AEB38055.SonarQubeAnalyze@4
-          displayName: 'Run Code Analysis'
+      - task: Gradle@2
+        displayName: 'Gradle build and test'
+        inputs:
+          workingDirectory: .
+          gradleWrapperFile: gradlew
+          tasks: 'clean build jacocoMergedReport'
+          publishJUnitResults: true
+          testResultsFiles: '**/TEST-*.xml'
+          sonarQubeRunAnalysis: true
+          sqGradlePluginVersionChoice: 'build'
 
-        - task: SonarQubePublish@4
-          displayName: 'SonarQube Publish Code Analysis'
-          inputs:
-            pollingTimeoutSec: 600
+      - task: PublishCodeCoverageResults@1
+        displayName: 'Publish code coverage'
+        inputs:
+          summaryFileLocation: '$(System.DefaultWorkingDirectory)/build/reports/jacocoMergedReport/jacocoMergedReport.xml'
 
-        - task: sonar-buildbreaker@8
-          displayName: 'Quality Gate Breaker'
-          inputs:
-            SonarQube: SonarQube
-          enabled: true
+      - task: Sonar-buildbreaker@8
+        displayName: 'Break Build on Quality Gate Failure'
+        inputs:
+          SonarQube: SonarQube
 
+      - task: PublishBuildArtifacts@1
+        displayName: 'Publish Artifact'
+        inputs:
+          PathtoPublish: '$(System.DefaultWorkingDirectory)/applications/app-service/build/libs/'
+          ArtifactName: '$(artifactName)'
+          publishLocation: 'Container'
+        condition: and(succeeded(), eq(variables['Build.SourceBranch'],'refs/heads/trunk'))
```

Comment:
Add comment to indicate the principal branch.

Reply:
TODO comments added